llvm/lib/Target/DirectX/DXILPostOptimizationValidation.cpp

```diff
@@ -118,6 +118,17 @@ static void reportOverlappingBinding(Module &M, DXILResourceMap &DRM) {
                        "true, yet no overlapping binding was found");
 }
 
+static void reportInvalidHandleTyError(Module &M, ResourceClass RC,
+                                       ResourceInfo::ResourceBinding Binding) {
+  SmallString<128> Message;
```

Comment:
The message here is always longer than 128 characters, so this SmallString is never able to use its "small" storage. We should bump this to something like 160 and avoid an unnecessary heap allocation